docs/contracts/v4/guides/11-position-manager.mdx

```diff
@@ -0,0 +1,838 @@
+
+## **CLOSE_CURRENCY: Handling Unknown Deltas**
+
+When you can’t predict whether you’ll need to pay or receive tokens, CLOSE_CURRENCY automatically handles either case.
+
+```solidity
+// Example scenario: Converting fees to liquidity
+bytes memory actions = abi.encodePacked(
+    Actions.INCREASE_LIQUIDITY,
+    Actions.CLOSE_CURRENCY  // Will automatically settle or take based on final delta
+);
+
+bytes[] memory params = new bytes[](2);
+
+// Parameters for INCREASE_LIQUIDITY
+params[0] = abi.encode(
+    tokenId,
+    liquidityIncrease,
+    amount0Max,
+    amount1Max,
+    ""
+);
+
+// CLOSE_CURRENCY only needs the currency
+params[1] = abi.encode(currency0);
+```
+
+This is particularly useful when:
+
+- Converting fees to liquidity (fees might fully cover the increase)
+- Complex operations where final deltas are uncertain
+- Reducing code complexity by letting the protocol handle the direction
+
+## **CLEAR_OR_TAKE: Optimizing for Dust**
+
+Sometimes receiving small token amounts costs more in gas than they’re worth. CLEAR_OR_TAKE lets you specify a threshold:
+
+```solidity
+// Parameters for CLEAR_OR_TAKE
+params[0] = abi.encode(
+    currency,     // The token to handle
+    threshold     // Minimum amount worth taking
+);
+```
+
+If the amount to receive is:
+
+- Above threshold: Tokens are taken (like TAKE_PAIR)
+- Below threshold: Amount is forfeited, saving gas
+
+This is valuable for:
+
+- Fee collection with minimum amounts
+- Operations where dust amounts can be ignored
+- Gas optimization in production systems
+
+## **SWEEP: Handling Excess Payments**
+
+SWEEP helps recover any excess tokens sent to the PoolManager:
+
+```solidity
+bytes memory actions = abi.encodePacked(
+    Actions.YOUR_MAIN_OPERATION,
+    Actions.SWEEP  // Add at the end to collect any excess
+);
+
+// Parameters for SWEEP
+params[1] = abi.encode(
+    currency,   // Token to sweep
+    recipient   // Where to send excess tokens
+);
+```
+
+Use SWEEP when:
+
+- Dealing with native ETH operations
+- Conservative token approvals might result in excess
+- Need to ensure all tokens are properly accounted for
+
+## **Understanding modifyLiquiditiesWithoutUnlock**
+
+This function follows the same encoding and command patterns as `modifyLiquidity`, but serves a specific purpose: it's used when the PoolManager is already unlocked. This is particularly useful in certain scenarios:
+
+- When called from hooks that are already executing within the PoolManager's lock/unlock cycle
+- For operations like automatic fee compounding, where a hook might want to reinvest fees for users
+
+For example, a hook that automatically compounds fees for users would use this function because the hook is already executing within the PoolManager's unlock context, there's no need for another unlock cycle and it's more efficient than calling the standard `modifyLiquidity`.
```

Comment:
```suggestion
For example, a hook that automatically compounds fees for users would use `modifyLiquiditiesWithoutUnlock` because the hook is already executing within the PoolManager's unlock context, and cannot re-unlock the PoolManager
```